app/src/main/jni/main.cpp

```diff
@@ -80,9 +81,17 @@ static void init_methods_cache(JNIEnv *env) {
     java_Boolean_init = env->GetMethodID(java_Boolean, "<init>", "(Z)V");
     java_Boolean_booleanValue = env->GetMethodID(java_Boolean, "booleanValue", "()Z");
     #undef FIND_CLASS
+    jclass java_GLSurfaceView = env->FindClass("android/opengl/GLSurfaceView");
+    java_GLSurfaceView_requestRender = env->GetMethodID(java_GLSurfaceView, "requestRender", "()V");
     methods_initialized = true;
 }
 
+static void *render_cb(void *data)
+{
+    jobject *glView = (void *)data;
+    return env->CallVoidMethod(glView, java_GLSurfaceView_requestRender);
```

Comment:
We don't have `env` here. The problem is that since this callback is executed from a different thread, you have to attach a java vm to it and obtain its own `env` (JNIEnv is valid only for current thread). Check out `android_content.cpp` for an example.